content/content.go

```diff
@@ -151,6 +151,18 @@ func WithLabels(labels map[string]string) Opt {
 	}
 }
 
+// WithExpiration sets an expiration on the lease
+func WithExpiration(d time.Duration) Opt {
```

Comment:
💯

Reply:
Changed the word lease here. Was copied and pasted from the lease implementation.

Reply:
This should be able to leverage leases directly. We don't need multiple types of expiring objects, leases were intended to the one object which could hold reference and expire. We are also adding more interface to add objects to leases directly in cases where there is no write to put an object into a lease or there is a secondary lease.

Reply:
This is based on https://github.com/containerd/containerd/pull/3304 yea? Is it likely to be merged soon and we can just update the code to use it?

Reply:
Cannot be based on it since it predates #3304...

Reply:
@dmcgowan implemented. https://github.com/containerd/containerd/pull/3134/commits/a93ccb1d47f6369ab70b2aef94d07a4f5764ce71